lib/seafoam/formatters/base.rb

```diff
@@ -0,0 +1,24 @@
+module Seafoam
+  module Formatters
+    module Base
+      # Base interface for all formatters. Returns a string representation of the associated command's output.
+      module Formatter
+        def format
+          raise NotImplementedError
+        end
+      end
```

Comment:
It’s not particularly “out there” but it’s also not doing anything for us yet. No big deal, either seems fine to me. (I know that some people feel more strongly about reserving `NotImplementedError` for specifically [“when a feature is not implemented on the current platform”](https://docs.ruby-lang.org/en/3.1/NotImplementedError.html) rather than missing methods in general, but I’m pretty sure MRI doesn’t care.)

Reply:
Interesting. I've never seen that distinction before. It's probably selection bias, but I've mostly seen it to mean "yes, this method does exist but you're either calling it on the wrong object or it's not implemented for this object" to avoid any confusion around `NoMethodError` not referring a typo.

I'll circle back to this afterwards. For now, I'm finding it helpful to distinguish between commands that haven't been updated for the formatter system at all and those that have, but haven't been fully implemented yet.